extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureDataSegmentKiller.java

```diff
@@ -72,9 +86,26 @@ public void kill(DataSegment segment) throws SegmentLoadingException
   }
 
   @Override
-  public void killAll()
+  public void killAll() throws IOException
   {
-    throw new UnsupportedOperationException("not implemented");
+    log.info("Deleting all segment files from Azure storage location [bucket: '%s' prefix: '%s']",
+             segmentConfig.getContainer(), segmentConfig.getPrefix()
+    );
+    try {
+      AzureUtils.deleteObjectsInPath(
+          azureStorage,
+          inputDataConfig,
+          accountConfig,
+          azureCloudBlobIterableFactory,
+          segmentConfig.getContainer(),
+          segmentConfig.getPrefix(),
+          Predicates.alwaysTrue()
+      );
+    }
+    catch (Exception e) {
+      log.error("Error occurred while deleting segment files from s3. Error: %s", e.getMessage());
```

Comment:
Heh, should probably be

```suggestion
      log.error("Error occurred while deleting segment files from Azure. Error: %s", e.getMessage());
```

Reply:
done